subsys/net/l2/canbus/canbus.c

```diff
@@ -1002,6 +1082,191 @@ static enum net_verdict process_frame(struct net_pkt *pkt)
 	return ret;
 }
 
+#ifdef CONFIG_NET_L2_CANBUS_ETH_TRANSLATOR
+static const u16_t mcast_addr = sys_cpu_to_be16(NET_CAN_MULTICAST_ADDR);
+
+static void eth_to_can_lladdr(struct net_pkt *pkt, bool bcast)
+{
+	struct net_eth_hdr *hdr = NET_ETH_HDR(pkt);
+	struct net_linkaddr *lladdr;
+
+	lladdr = net_pkt_lladdr_src(pkt);
+	lladdr->addr = hdr->src.addr;
+	lladdr->len = sizeof(struct net_eth_addr);
+	lladdr->type = NET_LINK_ETHERNET;
+
+	lladdr = net_pkt_lladdr_dst(pkt);
+	lladdr->len = sizeof(struct net_can_lladdr);
+	lladdr->type = NET_LINK_CANBUS;
+	if (bcast) {
+		lladdr->addr = (u8_t *)&mcast_addr;
+	} else {
+		lladdr->addr = &hdr->dst.addr[4];
+	}
+}
+
+static void forward_eth_frame(struct net_pkt *pkt)
+{
+	struct net_if *canbus_iface;
+
+	canbus_iface = net_if_get_first_by_type(&NET_L2_GET_NAME(CANBUS));
```

Comment:
Initially this is ok, but we need to figure out what to if there are multiple CAN interfaces in the system.